src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -349,5 +351,124 @@ public class SendReceiveTest : ClientWebSocketTestBase
+
+                    Assert.Equal(WebSocketState.Open, clientSocket.State);
+                    Assert.Null(clientSocket.CloseStatus);
+                }
+                else if (PlatformDetection.IsFullFramework)
+                {
+                    _output.WriteLine("[Windows] ManagedWebSocket-based implementation.");
+
+                    WebSocketException pendingReceiveException = await Assert.ThrowsAsync<WebSocketException>(() => pendingReceiveAsync);
+                    Assert.Equal(WebSocketError.ConnectionClosedPrematurely, pendingReceiveException.WebSocketErrorCode);
+
+                    WebSocketException newReceiveException =
+                        await Assert.ThrowsAsync<WebSocketException>(() => clientSocket.ReceiveAsync(recvSegment, cts.Token));
+                    Assert.Equal(WebSocketError.Success, newReceiveException.WebSocketErrorCode);
+                    Assert.Equal(
+                        ResourceHelper.GetExceptionMessage("net_WebSockets_InvalidState", "Aborted", "Open, CloseSent"),
+                        newReceiveException.Message);
+
+                    Assert.Equal(WebSocketState.Aborted, clientSocket.State);
+                    Assert.Null(clientSocket.CloseStatus);
+                }
+                else if (PlatformDetection.IsUap)
+                {
+                    _output.WriteLine("WinRTWebSocket-based implementation.");
+
+                    const uint WININET_E_CONNECTION_ABORTED = 0x80072EFE;
+
+                    WebSocketException pendingReceiveException = await Assert.ThrowsAsync<WebSocketException>(() => pendingReceiveAsync);
+                    Assert.Equal(WebSocketError.ConnectionClosedPrematurely, pendingReceiveException.WebSocketErrorCode);
+                    Assert.NotNull(pendingReceiveException.InnerException);
+                    Assert.Equal(WININET_E_CONNECTION_ABORTED, (uint)pendingReceiveException.InnerException.HResult);
+
+                    WebSocketException newReceiveException =
+                        await Assert.ThrowsAsync<WebSocketException>(() => clientSocket.ReceiveAsync(recvSegment, cts.Token));
+                    Assert.Equal(WebSocketError.Success, newReceiveException.WebSocketErrorCode);
+                    Assert.Equal(
+                        ResourceHelper.GetExceptionMessage("net_WebSockets_InvalidState", "Aborted", "Open, CloseSent"),
+                        newReceiveException.Message);
+
+                    Assert.Equal(WebSocketState.Aborted, clientSocket.State);
+                    Assert.Null(clientSocket.CloseStatus);
+                }
+                else
+                {
+                    _output.WriteLine("WinHttpWebSocket-based implementation.");
+
+                    const uint WININET_E_CONNECTION_RESET = 0x80072eff;
+
+                    Win32Exception pendingReceiveException = await Assert.ThrowsAnyAsync<Win32Exception>(() => pendingReceiveAsync);
```

Comment:
Why is this WinHttpWebSocket behavior being accepted and codified in a test rather than it being a bug?  All of the other implementations are throwing a WebSocketException, but this is throwing a Win32Exception?
cc: @davidsh

Reply:
My objective with this test was to pin the behaviors of the various WebSocketClient.ReceiveAsync implementations, though the WinHttpWebSocket behavior codified here is essentially a repro of #14103.

Reply:
Instead of this being a test, it should have been separated out and marked as ActiveIssue against #14103.